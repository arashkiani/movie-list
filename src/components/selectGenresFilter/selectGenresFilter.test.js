/*global shallow, mount, sinon, store, mockStore */
import {SelectGenresFilter} from './index'
import genres from '../../mocks/genres'

const mapToProps = {
	genres: genres.genres,
	selectedOptions:  [],
	getGenres: jest.fn(() => ({})),
	updateGenresFilter: jest.fn(({target}) => {
		mapToProps.selectedOptions.push(target.value)
	}),
}
let mountWrapper;
beforeEach(() => {
	mountWrapper = mount(<SelectGenresFilter {...store} {...mapToProps}/>)
})
afterEach(() => {
	mountWrapper.unmount();
})
describe('<SelectGenresFilter />', () => {
	it('Match Snapshot', () => {
		expect(mountWrapper).toMatchSnapshot()
	})
	it('User can toggle button to get a list of Genres', () => {
		const container = mountWrapper.find('button').first()
		expect(container.length).toBe(1)
		container.simulate('click')
		expect(mountWrapper.find('label').length).toBe(genres.genres.length)
	})
	it('Select Genre option from the list', () => {
		mountWrapper.find('button').first().simulate('click')
		const value = mountWrapper.find('input').first().props().defaultValue
		const SelectGenre = mountWrapper.find('form').first()
		SelectGenre.simulate('change', { target: { value } })
		expect(mountWrapper.props().selectedOptions[0]).toBe(value)
	})
})