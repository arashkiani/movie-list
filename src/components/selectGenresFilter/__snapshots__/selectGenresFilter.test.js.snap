// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectGenresFilter /> Match Snapshot 1`] = `
<div>
  <ForwardRef
    onClick={[Function]}
  >
    <button
      className="sc-htpNat jSUnNu"
      onClick={[Function]}
    >
      Filter 
      â–¼
    </button>
  </ForwardRef>
</div>
`;
